### PR TITLE
[RDY] tests: migrate legacy charsearch test.

### PR DESCRIPTION
**Pull request by lucc**
Part of #1286.

The commit 9cbc8e9 is very small and could be squashed but I would keep it to "document" the change.

### CODE REVIEW

File: test/functional/legacy/charsearch_spec.lua [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    execute('/^X')
+    feed('ylfep;;p,,p')
+    -- Check that save/restore works.
+    execute('/^Y')
+    feed('ylfep')
+    execute('let csave = getcharsearch()')
+    feed('fip')
+    execute('call setcharsearch(csave)')
+    feed(';p;p')
+    -- Check that setcharsearch() changes the settins.
```
Comment (justinmk) [inline test/functional/legacy/charsearch_spec.lua]: typo

File: test/functional/legacy/charsearch_spec.lua [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    execute('/^Z')
+    feed('ylfep')
+    execute("call setcharsearch({'char': 'k'})")
+    feed(';p')
+    execute("call setcharsearch({'forward': 0})")
+    feed('$;p')
+    execute("call setcharseearch({'until': 1})")
```
Comment (justinmk) [inline test/functional/legacy/charsearch_spec.lua]: very weird: original test also called `setcharseearch` which is not a valid function name.

File: test/functional/legacy/charsearch_spec.lua (hunk 1 above)
Comment (lucc) [inline test/functional/legacy/charsearch_spec.lua]: Will fix, this is also part of vim 7.4.1041.

File: test/functional/legacy/charsearch_spec.lua (hunk 2 above)
Comment (lucc) [inline test/functional/legacy/charsearch_spec.lua]: I didn't even notice this typo in the original test. If I correct it the test fails. 

Is this a vim bug? Could you ask on the vim mailing list (James McCoy wrote the patch 7.4.813 for these functions). Or should we just fix it somehow or leave it?

(I'm not on the vim list that's why I ask.)

Comment (jamessan) [inline test/functional/legacy/charsearch_spec.lua]: I didn't actually write the tests.  Bram did that when he included my patch and that part of the test is buggy.  The expected output is based on `{'until': 0}`, which is why the tests succeed even though the `:call` fails.

Comment (jamessan) [inline test/functional/legacy/charsearch_spec.lua]: I've created vim/vim#650 to fix this upstream.

Comment (lucc) [inline test/functional/legacy/charsearch_spec.lua]: Should I use the version which is in upstream vim at this moment or already include the patch from vim/vim#650?

**Comment by justinmk** [comment]
@lucc please include the fix

**Comment by lucc** [comment]
I assume you did a full review when you found the error discussed above, so I think it is RDY.

**Comment by justinmk** [comment]
Merged after updating version.c. Thanks @lucc and @jamessan

**Pull request closed without merging**